common/shell/shell_platform.c

```diff
@@ -0,0 +1,607 @@
+            * Set INT type      Set one gpio interrupt T    o           platform gpio set int_type <gpio_num> <type>
+        * SENSOR                                            O
+            * List all          List all sensors' info      o           platform sensor list_all
+            * Get               Get one sensor info         o           platform sensor get <sensor_num>
+            * Set polling en    Set one sensor polling      x           TODO
+            * Set mbr           Set one sensor mbr          x           TODO
+            * Set threshold     Set one sensor threshold    x           TODO
+        * I2C SLAVE                                         X
+
+    (2) Some hard code features need to be modified if CHIP is different
+
+    (3) GPIO List all/Get/Set value/Set direction are not protected by PINMASK_RESERVE_CHECK. It might cause system-hang problem!
+
+    (4) "List multifnctl" command only lists relative registers mentioned in chip spec
+        chapter "5.Multi-function Pins Mapping and Control" 
+*/
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <logging/log.h>
+#include <shell/shell.h>
+#include <device.h>
+#include <devicetree.h>
+
+/* Include GPIO */
+#include <drivers/gpio.h>
+#include "plat_gpio.h"
+#include "hal_gpio.h"
+
+/* Include SENSOR */
+#include "sensor.h"
+
+/* Include config settings */
+#include "shell_platform.h"
+
+/* Declare Log */
+LOG_MODULE_REGISTER(shell_platform);
+
+/* Declare Code info */
+#define RELEASE_VERSION "v1.0.0"
+#define RELEASE_DATE "2022.01.11"
+
+/*********************************************************************************************************
+ * TOOL SECTION
+**********************************************************************************************************/
+/* 
+    Command GPIO 
+*/
+#if COMMAND_GPIO == 1
+#if PINMASK_RESERVE_CHECK
+enum CHECK_RESV {
+    CHECK_BY_GROUP_IDX,
+    CHECK_BY_GLOBAL_IDX
+};
+static int gpio_check_reserve(const struct device *dev, int gpio_idx, enum CHECK_RESV mode) 
+{
+    if (!dev)
+        return 1;
+    
+    const struct gpio_driver_config *const cfg = (const struct gpio_driver_config *)dev->config;
+
+    if (!cfg->port_pin_mask)
+        return 1;
+
+    int gpio_idx_in_group;
+    if (mode == CHECK_BY_GROUP_IDX) {
+        gpio_idx_in_group = gpio_idx;
+    } else if (mode == CHECK_BY_GLOBAL_IDX) {
+        gpio_idx_in_group = gpio_idx%GPIO_GROUP_SIZE;
+    } else
+        return 1;
+
+    if ( (cfg->port_pin_mask & (gpio_port_pins_t)BIT(gpio_idx_in_group) ) == 0U)
+        return 1;
+
+    return 0;
+}
+#endif
+
+static int gpio_access_cfg(const struct shell *shell, int gpio_idx, enum GPIO_ACCESS mode, int *data)
+{
+    if (!shell)
+        return 1;
+
+    if (gpio_idx >= GPIO_CFG_SIZE || gpio_idx < 0)
+        return 1;
+
+    /* check whether reserve pin defined in pin NAME */
```

Comment:
Any need to keep this commented code?  On the other hand, it might make sense to enable it to get rid of so many #if statements elsewhere.